gpt_index/playground/base.py

```diff
@@ -0,0 +1,116 @@
+"""Experiment with different indices, models, and more."""
+from __future__ import annotations
+
+import time
+from typing import Any, Dict, List, Optional, Union
+
+import pandas as pd
+from langchain.input import get_color_mapping, print_text
+
+from gpt_index.indices.base import BaseGPTIndex
+from gpt_index.indices.list.base import GPTListIndex
+from gpt_index.indices.tree.base import GPTTreeIndex
+from gpt_index.indices.vector_store.simple import GPTSimpleVectorIndex
+from gpt_index.readers.schema.base import Document
+
+DEFAULT_INDEX_CLASSES = [GPTSimpleVectorIndex, GPTTreeIndex, GPTListIndex]
+DEFAULT_MODES = ["default", "summarize", "embedding", "retrieve", "recursive"]
+
+
+class Playground:
+    """Experiment with indices, model, embeddings, modes, and more."""
+
+    def __init__(
+        self, indices: List[BaseGPTIndex], modes: Optional[List[str]] = DEFAULT_MODES
+    ):
+        """Initialize with indices to experiment with.
+
+        Args:
+            indices: A List of BaseGPTIndex's to experiment with
+        """
+        self.update_indices(indices)
```

Comment:
the term `update` is a big confusing, think for this case it's more canonical to define a setter https://www.geeksforgeeks.org/getter-and-setter-in-python/. You can abstract out the validation logic in a `_validate_indices` method (same with modes)

Reply:
Oh cool